Gems/SceneProcessing/Code/Source/Generation/Components/TangentGenerator/TangentGenerateComponent.cpp

```diff
@@ -191,95 +184,175 @@ namespace AZ::SceneGenerationComponents
+                AZ_TracePrintf(AZ::SceneAPI::Utilities::ErrorWindow, "Cannot generate tangents for uv set %zu as it cannot be retrieved.\n", uvSetIndex);
+                continue;
+            }
+
+            // Check if we had tangents inside the source scene file.
+            AZ::SceneAPI::DataTypes::IMeshVertexTangentData* tangentData = FindTangentData(graph, nodeIndex, uvSetIndex);
+            AZ::SceneAPI::DataTypes::IMeshVertexBitangentData* bitangentData = FindBitangentData(graph, nodeIndex, uvSetIndex);
+
+            // If all we need is import from the source scene, and we have tangent data from the source scene already, then skip generating.
+            if ((tangentSpace == AZ::SceneAPI::DataTypes::TangentSpace::FromSourceScene))
+            {
+                if (tangentData && bitangentData)
                 {
-                // If we want Fbx tangents, we don't need to do anything for that.
-                case AZ::SceneAPI::DataTypes::TangentSpace::FromSourceScene:
+                    AZ_TracePrintf(AZ::SceneAPI::Utilities::LogWindow, "Using source scene tangents and bitangents for uv set %zu for mesh '%s'.\n",
+                        uvSetIndex, scene.GetGraph().GetNodeName(nodeIndex).GetName());
+                    continue;
+                }
+                else
                 {
-                    allSuccess &= true;
+                    // In case there are no tangents/bitangents while the user selected to use the source ones, default to MikkT.
+                    AZ_TracePrintf(AZ::SceneAPI::Utilities::WarningWindow, "Cannot use source scene tangents as there are none in the asset for mesh '%s' for uv set %zu. Defaulting to generating tangents using MikkT.\n",
+                        scene.GetGraph().GetNodeName(nodeIndex).GetName(), uvSetIndex);
+                    tangentSpace = AZ::SceneAPI::DataTypes::TangentSpace::MikkT;
                 }
-                break;
+            }
 
-                // Generate using MikkT space.
-                case AZ::SceneAPI::DataTypes::TangentSpace::MikkT:
+            if (!tangentData)
+            {
+                if (!AZ::SceneGenerationComponents::TangentGenerateComponent::CreateTangentLayer(scene.GetManifest(), nodeIndex, meshData->GetVertexCount(), uvSetIndex,
+                    tangentSpace, graph, &tangentData))
                 {
-                    allSuccess &= AZ::TangentGeneration::Mesh::MikkT::GenerateTangents(
-                        scene.GetManifest(), graph, nodeIndex, const_cast<AZ::SceneAPI::DataTypes::IMeshData*>(meshData), uvSetIndex);
-                    for (AZ::SceneData::GraphData::BlendShapeData* blendShape : blendShapes)
-                    {
-                        allSuccess &= AZ::TangentGeneration::BlendShape::MikkT::GenerateTangents(blendShape, uvSetIndex);
-                    }
+                    AZ_TracePrintf(AZ::SceneAPI::Utilities::ErrorWindow, "Failed to create tangents data set for mesh %s for uv set %zu.\n",
```

Comment:
Same here. In the az standard coding practice
AZ_TracePrintF - "Programmers might find this useful/It is possible something is incorrect"
AZ_Error - "Somethings wrong with the data.  We'll continue, but certain stuff is broken.  Generally, whoever made the offending file / asset / etc must fix it, but it won't require a code change to fix"
AZ_Warning - "Everything should still function, but sub-optimally.  It won't require a code change to fix, its a data problem"

I'd argue this is more of an AZ_Error?